### PR TITLE
week 6 assessment completed

### PR DESCRIPTION
**Pull request by OnyxFire1**
# Submitting Week 6 Assessment

### ✅ Completed Files
Select the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by OnyxFire1** [comment]
I was having difficulties getting the test to pass in code-challenge number #1, even with the help of my tutor.

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,38 +14,116 @@
...
+describe ("namesCap", () => {
+  it("takes an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    const desc = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }]
+    expect(namesCap(desc)).toEqual("Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee.")
+  })
+})
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:26]: good structure and failure

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,38 +14,116 @@
...
+// Use .reduce to loop through each step of the array and sum of all the previous steps
+
+const cubed = (arr) => {
+  let cubedArr = arr.map((value) => {
+    return value ** 3;
+  });
+  return cubedArr.reduce((a, b) => a + b, 0);
+};
+
+// PASS  ./code-challenges.test.js
```
Comment (SunkissedQueen) [inline code-challenges.test.js:127]: Passing again

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,38 +14,116 @@
...
+       numArray[i] = numArray[i] % 3;
+       }
+       return numArray
+     }
+
+// mixData
+
+//     ✓ that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (1 ms)
+
+// Test Suites: 1 passed, 1 total
```
Comment (SunkissedQueen) [inline code-challenges.test.js:92]: woohoo

File: code-challenges.test.js [hunk 4]
```diff
@@ -14,38 +14,116 @@
...
+  const namesCap = (array) => {
+    return array.map(newSentence => {
+      let name = newSentence.name.split(" ").map(value => value.replace(value[0], value[0].toUpperCase()))
+      let fullname = name.join(" ")
+      let desc = desc.occupation
+      let newSentence = `${fullname} is ${desc}.`
+      return newSentence
+    })
+  }
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:54]: You are so close. If you do corrections according to the error code, you will be able to pass.

Error messages:
`SyntaxError: /Users/charleanbaxter/Desktop/learn-2022-instruction/lecture-2022/react-state-cb/assessment.test.js: Identifier 'newSentence' has already been declared. (272:19)`
Correction ----> Changed the variable to a different name

 let sentence = `${fullname} is ${desc}.`
 return sentence

`ReferenceError: Cannot access 'desc' before initialization`
Correction ----> change the reference to the parameter designated on your function

let desc = newSentence.occupation

Further inspection shows that the expected outcome on the test was inputted without the square brackets to show the array.

    expect(namesCap(desc)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])

After the update, you passed. Follow the error messages, they are your friends.

`namesCap
    ✓ takes an array of objects and returns an array with a sentence about each person with their name capitalized (1 ms)`

File: interview-questions.md [hunk 5]
```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-  Your answer:
+  Your answer: To fix this mistake see method below. The name of the foreign key would be has_many students and is on the student model.
 
   Researched answer:
 
+  To correct the mistake: Step 1: run the command 'rails generate migration', add a new column to the table. Step 2: add_column :students, :cohort_id, :integer' Step 3: lastly, in terminal, run 'rails db:migrate'.
```
Comment (SunkissedQueen) [inline interview-questions.md:15]: Look over this section of the syllabus
https://github.com/learn-academy-2022-charlie/Syllabus/blob/main/rails/associations.md

File: interview-questions.md [hunk 6]
```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. Which RESTful routes must always be passed params? Why?
```
Comment (SunkissedQueen) [inline interview-questions.md:17]: https://github.com/learn-academy-2022-charlie/Syllabus/blob/main/rails/restful-routes-crud.md
Look at the paths that require an id

File: interview-questions.md [hunk 7]
```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
+  RAILS GENERATE controller - The logical center of your application. It coordinates the interaction between the user, the views, and the model.
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+  RAILS GENERATE migration - Migrations are a feature of Active Record that allows you to evolve your database schema over time. Rather than write schema modifications in pure SQL, migrations allow you to use a Ruby DSL to describe changes to your tables. 
 
-action: "GET"    location: /students          
 
-action: "POST"   location: /students       
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
```
Comment (SunkissedQueen) [inline interview-questions.md:41]: Key things to remember are that edit and new will give you a form to fill out. Edit is the form for update and new is the form for create. Just how I remembered it. 'Get' will display something be it everything (index), one thing (show), a form to create a new entry (new), a form to edit/update (edit).  'Post' will create something new (create). 'Patch/put' will update something that is already existing (update). Delete will delete something (destroy).

File: interview-questions.md [hunk 8]
```diff
@@ -7,44 +7,78 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-action: "DELETE" location: /students/2      
+action: "GET" location: /students/2/edit - Edit controller method, show edit form for one student.   
 
+action: "PATCH" location: /students/2 -  Update controller method, update particular students, then redirect.     
 
+action: "DELETE" location: /students/2 - Destroy controller method, delete student that redirect.     
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
```
Comment (SunkissedQueen) [inline interview-questions.md:58]: Good way to break it down to understand it

**Review by SunkissedQueen — APPROVED** [review]
Sean, you made it to the last assessment. Continue to practice. The searching will become easier as well as the ability to flow from one idea to the next. Drive and navigate; don't hesitate on your opportunity to grow. I left some notes on the first challenge question. Following error messages is a great tool to have as a web developer because we are human and can make mistakes. The ability to dig out of them is very beneficial. Remember to always use the interview questions to share how you have used that tool. Active learner, you are making progress.

**Pull request merged**